src/plugins/data/common/search/expressions/kibana_context.ts

```diff
@@ -40,75 +48,102 @@ const mergeQueries = (first: Query | Query[] = [], second: Query | Query[]) =>
     (n: any) => JSON.stringify(n.query)
   );
 
-export const kibanaContextFunction: ExpressionFunctionKibanaContext = {
-  name: 'kibana_context',
-  type: 'kibana_context',
-  inputTypes: ['kibana_context', 'null'],
-  help: i18n.translate('data.search.functions.kibana_context.help', {
-    defaultMessage: 'Updates kibana global context',
-  }),
-  args: {
-    q: {
-      types: ['kibana_query', 'null'],
-      aliases: ['query', '_'],
-      default: null,
-      help: i18n.translate('data.search.functions.kibana_context.q.help', {
-        defaultMessage: 'Specify Kibana free form text query',
-      }),
-    },
-    filters: {
-      types: ['kibana_filter', 'null'],
-      multi: true,
-      help: i18n.translate('data.search.functions.kibana_context.filters.help', {
-        defaultMessage: 'Specify Kibana generic filters',
-      }),
+export const getKibanaContextFn = (
+  getStartDependencies: (
+    getKibanaRequest: ExecutionContext['getKibanaRequest']
+  ) => Promise<KibanaContextStartDependencies>
+) => {
+  const kibanaContextFunction: ExpressionFunctionKibanaContext = {
+    name: 'kibana_context',
+    type: 'kibana_context',
+    inputTypes: ['kibana_context', 'null'],
+    help: i18n.translate('data.search.functions.kibana_context.help', {
+      defaultMessage: 'Updates kibana global context',
+    }),
+    args: {
+      q: {
+        types: ['kibana_query', 'null'],
+        aliases: ['query', '_'],
+        default: null,
+        help: i18n.translate('data.search.functions.kibana_context.q.help', {
+          defaultMessage: 'Specify Kibana free form text query',
+        }),
+      },
+      filters: {
+        types: ['kibana_filter', 'null'],
+        multi: true,
+        help: i18n.translate('data.search.functions.kibana_context.filters.help', {
+          defaultMessage: 'Specify Kibana generic filters',
+        }),
+      },
+      timeRange: {
+        types: ['timerange', 'null'],
+        default: null,
+        help: i18n.translate('data.search.functions.kibana_context.timeRange.help', {
+          defaultMessage: 'Specify Kibana time range filter',
+        }),
+      },
+      savedSearchId: {
+        types: ['string', 'null'],
+        default: null,
+        help: i18n.translate('data.search.functions.kibana_context.savedSearchId.help', {
+          defaultMessage: 'Specify saved search ID to be used for queries and filters',
+        }),
+      },
     },
-    timeRange: {
-      types: ['timerange', 'null'],
-      default: null,
-      help: i18n.translate('data.search.functions.kibana_context.timeRange.help', {
-        defaultMessage: 'Specify Kibana time range filter',
-      }),
+
+    extract(state) {
+      const references: SavedObjectReference[] = [];
+      if (state.savedSearchId.length && typeof state.savedeSearchId[0] === 'string') {
```

Comment:
The first clause is redundant because the `typeof` will return undefined for an unexisting item.

```suggestion
      if (typeof state.savedeSearchId[0] === 'string') {
```